src/ln/channelmanager.rs

```diff
@@ -447,6 +451,15 @@ macro_rules! break_chan_entry {
 				}
 				break Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
 			},
+			Err(ChannelError::Stale { msg, update}) => {
+				if let Some(update) = update {
+					if let Err(_e) = $self.monitor.add_update_monitor(update.get_funding_txo().unwrap(), update) {
+						unimplemented!();
+					}
+				}
+				log_error!($self, "Channel {} need to stale due to {}", log_bytes!($entry.key()[..]), msg);
+				break Err(MsgHandleErrInternal::from_chan_no_close(ChannelError::Stale { msg, update: None }, $entry.key().clone()))
```

Comment:
Huh? Shouldn't the channel get removed? Right now it just continues like its no problem?

Reply:
Panic now on this branch we should never be taken